ynh_install_python/ynh_install_python

```diff
@@ -0,0 +1,151 @@
+#!/bin/bash
+
+# Need also the helper https://github.com/YunoHost-Apps/Experimental_helpers/blob/master/ynh_handle_getopts_args/ynh_handle_getopts_args
+
+pyenv_install_dir="/opt/pyenv"
+# PYENV_ROOT is the directory of pyenv, it needs to be loaded as a environment variable.
+export PYENV_ROOT="$pyenv_install_dir"
+# Required dependencies
+pyenv_dependencies="build-essential libssl1.0-dev|libssl-dev zlib1g-dev libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm libncurses5-dev libncursesw5-dev xz-utils tk-dev libffi-dev liblzma-dev python-openssl git"
+
+# Install python version management
+#
+# [internal]
+#
+# usage: ynh_install_pyenv
+ynh_install_pyenv () {
+  echo "Installation of pyenv - python version management" >&2
+  # Build an app.src for pyenv
+  mkdir -p "../conf"
+  echo "SOURCE_URL=https://github.com/pyenv/pyenv/archive/v1.2.19.tar.gz
+SOURCE_SUM=e93466735ac9c34d68b7d5d71f32c16a2b4b1a6a1adffb85acc4126a3398b9d0" > "../conf/pyenv.src"
+  # Download and extract pyenv
+  ynh_setup_source "$pyenv_install_dir" pyenv
+
+  # Build an app.src for pyenv-virtualenv
+  mkdir -p "../conf"
+  echo "SOURCE_URL=https://github.com/pyenv/pyenv-virtualenv/archive/v1.1.5.tar.gz
+SOURCE_SUM=27ae3de027a6f6dccdca4085225512e559c6b94b31625bd2b357a18890a1e618" > "../conf/pyenv-virtualenv.src"
+  # Download and extract pyenv-virtualenv
+  ynh_setup_source "$pyenv_install_dir/plugins/pyenv-virtualenv" pyenv-virtualenv
+
+  (cd $pyenv_install_dir
+  ./src/configure && make -C src)
+
+# Create shims directory if needed
+if [ ! -d $pyenv_install_dir/shims ] ; then
+  mkdir $pyenv_install_dir/shims
+fi
+}
+
+# Install a specific version of python
+#
+# ynh_install_python will install the version of python provided as argument by using pyenv.
+#
+# pyenv (python version management) stores the target python version in a .python_version file created in the target folder (using pyenv local <version>)
+# It then uses that information for every python user that uses pyenv provided python command
+#
+# This helper creates a /etc/profile.d/pyenv.sh that configures PATH environment for pyenv
+# for every LOGIN user, hence your user must have a defined shell (as opposed to /usr/sbin/nologin)
+#
+# Don't forget to execute python-dependent command in a login environment
+# (e.g. sudo --login option)
+# When not possible (e.g. in systemd service definition), please use direct path
+# to pyenv shims (e.g. $PYENV_ROOT/shims/bundle)
+#
+# usage: ynh_install_python python_version user
+# | arg: -v, --python_version= - Version of python to install.
+#        If possible, prefer to use major version number (e.g. 8 instead of 8.10.0).
```

Comment:
I'm not sure from the example `(e.g. 8 instead of 8.10.0)` if major version number in Python is 3 or 3.7.
We may want to specify the example with Python numbering...

Reply:
yes... My bad, it's a bad copy/pasta from ynh_install_ruby